File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/character/stats/LifePointsRegenBeginMessage.scala





















// Generated on 05/03/2013 10:34:37
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait LifePointsRegenBeginMessage extends MessageDeserializer {
  val messageId = 5684

  protected trait Message extends BaseMessage with Serializable {
    val messageId = LifePointsRegenBeginMessage.messageId

    val regenRate: UByte

    def serialize(buf: Buffer) {
      buf.writeUByte(regenRate)
    }

  }

  protected def create(regenRate: UByte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val regenRate = buf.readUByte()

    create(regenRate)
  }

}

object LifePointsRegenBeginMessage extends LifePointsRegenBeginMessage {
  case class MessageImpl(regenRate: UByte) extends Message
  type Target = MessageImpl

  override def create(regenRate: UByte) = MessageImpl(regenRate)

  def apply(regenRate: UByte) = create(regenRate)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/LifePointsRegenBeginMessage.java


// Generated on 05/08/2013 19:37:42
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class LifePointsRegenBeginMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5684;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short regenRate;
    
    public LifePointsRegenBeginMessage() { }
    
    public LifePointsRegenBeginMessage(short regenRate) {
        this.regenRate = regenRate;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUByte(regenRate);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        regenRate = buf.readUByte();
        if (regenRate < 0 || regenRate > 255)
            throw new RuntimeException("Forbidden value on regenRate = " + regenRate + ", it doesn't respect the following condition : regenRate < 0 || regenRate > 255");
    }
    
}
